### PR TITLE
Migrated to cookies

### PR DESCRIPTION
**Pull request by oesmith**
As mentioned on the discussion thread for issue #8, here's my small patch to switch to cookies.

Cheers!

Olly

### CODE REVIEW

File: lib/twitter.js [hunk 1]
```diff
@@ -248,12 +255,11 @@ Twitter.prototype.stream = function(method, params, callback) {
 /*
  * TWITTER "O"AUTHENTICATION UTILITIES, INCLUDING THE GREAT
  * CONNECT/STACK STYLE TWITTER "O"AUTHENTICATION MIDDLEWARE
- * and helpful utilities to retrieve the twauth cookie etc.
  */
-Twitter.prototype.cookie = function(req) {
+Twitter.prototype.cookie = function(cookies) {
```
Comment (jdub) [inline lib/twitter.js]: This API change makes it less of a drop-in replacement. If you can work around this, I'll pull this in as a medium-term change (as the OAuth web helper stuff will eventually move into another module).

**Comment by jdub** [comment]
Thanks heaps! If we can avoid the API changes, I'll pull this in straight away. :-)

File: lib/twitter.js [hunk 2]
```diff
@@ -51,12 +53,17 @@ function Twitter(options) {
...
+		new Keygrip([this.options.cookie_secret]);
+
 	this.oauth = new oauth.OAuth(
 		this.options.request_token_url,
 		this.options.access_token_url,
 		this.options.consumer_key,
 		this.options.consumer_secret,
-		'1.0', null, 'HMAC-SHA1', null,
+		'1.0A',
+		this.options.callback_url,
```
Comment (jdub) [inline lib/twitter.js:65]: Good call. :-)

**Comment by oesmith** [comment]
Thanks lots for the feedback - much appreciated.

I've refactored to put the API back to how it was - all in the latest commit.

Cheers!

Olly

**Comment by oesmith** [comment]
Whoops.. didn't mean to hit 'Close'..

File: lib/twitter.js [hunk 3]
```diff
@@ -248,16 +255,11 @@ Twitter.prototype.stream = function(method, params, callback) {
 /*
  * TWITTER "O"AUTHENTICATION UTILITIES, INCLUDING THE GREAT
  * CONNECT/STACK STYLE TWITTER "O"AUTHENTICATION MIDDLEWARE
- * and helpful utilities to retrieve the twauth cookie etc.
```
Comment (jdub) [inline lib/twitter.js:251]: This should not be removed.

File: lib/twitter.js [hunk 4]
```diff
@@ -291,7 +289,8 @@ Twitter.prototype.login = function(mount, success) {
 		}
 
 		// Fetch the cookie
-		var twauth = self.cookie(req);
```
Comment (jdub) [inline lib/twitter.js:294]: No need to replace this with the return of self.cookie()

Comment (oesmith) [inline lib/twitter.js:294]: 'cookies' needs to be in scope lower down to set new cookie values.

I'm not sure if there's an more elegant way to do this with calling self.cookie(), without changing its signature.  What do you think?

File: lib/twitter.js (hunk 3 above)
Comment (oesmith) [inline lib/twitter.js:251]: Have re-instated this with latest commit.

File: lib/twitter.js (hunk 4 above)
Comment (jdub) [inline lib/twitter.js:294]: Ah yes, you need res to set... luckily, signature matters not. :-)

If prototype.cookie took (req, res), and passed res || null to the Cookies constructor, we'd be fine.

OK with that? Thanks!

Comment (oesmith) [inline lib/twitter.js:294]: We still need to Cookies object though.. how would you get that back from prototype.cookie?

**Comment by fat** [comment]
what's the status of this... why not merge? Getting this message is disconcerting:

```
This library has been deprecated. Please use `npm install cookies` instead.
```

**Comment by oesmith** [comment]
fwiw, I've been using this patch in production on http://tweetgp.com for the last 3 months with no issues.

**Comment by neopunisher** [comment]
same message as fat... please update

**Comment by aseemk** [comment]
+1

**Comment by kc-dot-io** [comment]
+2

**Comment by masterkain** [comment]
is this fork totally abandoned?

**Comment by oesmith** [comment]
I'm still using my fork in production.

Whether or not the changes will get pulled back into jdub's repo is another matter though..

**Pull request merged**